contracts/game.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract Game is Ownable {
+  IERC20 private _token;
+  address [] judges;
+  address payable[] playersSeekingVerification;
+  address payable[] playersVerifiedToWin;
+  address payable[] winners;
+
+  event PlayerSeeksVerification(address indexed _player);
+  event PlayerVerifiedToWin(address indexed _player, address indexed _judge);
+  event PlayerWon(address indexed player);
+
+  constructor(IERC20 token) {
+    _token = token;
+  }
+
+  modifier judgeOnly {
+    require(isJudge(msg.sender), "Must be a judge");
+    _;
+  }
+
+  function isJudge(address judge) view public returns (bool) {
+    bool contains = false;
+    for (uint i = 0; i < judges.length; i++) {
+      if (judge == judges[i]) {
+        contains = true;
+      }
+    }
+    return contains;
+  }
+
+  function hasTokens(address player) view public returns (bool) {
+    return _token.balanceOf(player) >= 100 * 10**18;
+  }
+
+  function hasTooManyTokens(address player) view public returns (bool) {
+    return _token.balanceOf(player) >= 1000 * 10**18;
+  }
+
+  function hasBeenVerifiedToWin(address player) view public returns (bool) {
+    bool contains = false;
+    for (uint i = 0; i < playersVerifiedToWin.length; i++) {
+      if (player == playersVerifiedToWin[i]) {
+        contains = true;
+      }
+    }
+    return contains;
+  }
+
+  function hasNotWonBefore(address player) view public returns (bool) {
+    return !hasWon(player);
+  }
+
+  function seekVerification() public {
+    require(hasTokens(msg.sender), "Need 100 FWEB3 tokens");
+    require(!hasTooManyTokens(msg.sender), "Need less than 1,000 FWEB3 tokens");
+    playersSeekingVerification.push(payable(msg.sender));
+    emit PlayerSeeksVerification(msg.sender);
+  }
+
+  function win() public {
+    require(hasTokens(msg.sender), "Need 100 FWEB3 tokens");
+    require(!hasTooManyTokens(msg.sender), "Need less than 1,000 FWEB3 tokens");
+    require(hasBeenVerifiedToWin(msg.sender), "Need to have been verified by a judge");
+    require(hasNotWonBefore(msg.sender), "Need to have not won before");
+    (bool sent, ) = (msg.sender).call{ value: 1000 * (10*18) }("");
+    require(sent, "Failed to send Ether");
+    winners.push(payable(msg.sender));
+    emit PlayerWon(msg.sender);
+  }
+
+  function approveWinner(address player) public judgeOnly {
+    removePlayerFromSeekingVerification(player);
+    playersVerifiedToWin.push(payable(player));
+    emit PlayerVerifiedToWin(player, msg.sender);
+  }
+
+  function rejectWinner(address player) public {
```

Comment:
Need to add some logic around this so not anyone can just reject a winner (should be judgeOnly).

But more importantly, what’s the intention here? If one judge decides not to approve someone, they can reject them from the competition and they have to start over?

Reply:
for now they're just removed from the list, so that "spam" doesn't clog up a dashboard we build